cgi-bin/DW/Controller/Support/History.pm

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/perl
+#
+# DW::Controller::Support::History
+#
+# This controller is for the Support History page.
+#
+# Authors:
+#      hotlevel4 <hotlevel4@hotmail.com>
+#
+# Copyright (c) 2015 by Dreamwidth Studios, LLC.
+#
+# This is based on code originally implemented on LiveJournal.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself.  For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+
+package DW::Controller::Support::History;
+
+use strict;
+use warnings;
+
+use DW::Controller;
+use DW::Routing;
+use DW::Template;
+
+DW::Routing->register_string( '/support/history', \&history_handler, app => 1 );
+
+sub history_handler {
+    my $r = DW::Request->get;
+    my $get = $r->get_args;
+
+    my ( $ok, $rv ) = controller( anonymous => 0, form_auth => 1 );
+    return $rv unless $ok;
+    
+    my $vars = {};    
+
+    my $remote = $rv->{remote};
+    my $fullsearch = $remote->has_priv( 'supporthelp' );
+
+    $vars->{fullsearch} = $fullsearch;
+    
+    my $ret;
+    
+    if ( $get->{user} || $get->{email} || $get->{userid} ) {
+        my $dbr = LJ::get_db_reader();
+        return error_ml( '/support/history.tt.error.nodatabase' ) unless $dbr;
+        
+        $vars->{get_user} = ( $get->{user} ) ? 1 : 0;
+        $vars->{get_userid} = ( $get->{userid} ) ? 1 : 0;
+        $vars->{get_email} = ( $get->{email} ) ? 1 : 0;
+        $vars->{user} = $remote->user;
+                
+        my $reqlist;
+        if ( $get->{user} || $get->{userid} ) {
+            # get requests by a user, regardless of email (only gets user requests)
+            my $userid = $get->{userid} ? $get->{userid}+0 : LJ::get_userid( LJ::trim( $get->{user} ) );
+            return error_ml( '/support/history.tt.error.invaliduser' ) unless $userid
+                && ( $fullsearch || $remote->id == $userid );
+            $vars->{username} = LJ::ljuser( LJ::get_username( $userid ) );
+            $reqlist = $dbr->selectall_arrayref( 
+                'SELECT spid, subject, state, spcatid, requserid, ' .
+                       'timecreate, timetouched, timelasthelp, reqemail ' .
+                'FROM support WHERE reqtype = \'user\' AND requserid = ?',
+                undef, $userid );
+        } elsif ( $get->{email} ) {
+            # try by email, note that this gets requests opened by users and anonymous
+            # requests, so we can view them all
+            my $email = LJ::trim( $get->{email} );
+            $vars->{email} = $email;
```

Comment:
You dropped an important `LJ::ehtml` call here.  User input must always be escaped before being redisplayed.
```
$vars->{email} = LJ::ehtml( $email );
```
It's also possible to do the html escaping in the template, but it's probably simpler to do it here.